File: src/backend/schemas/review.py
"""리뷰 관련 API 스키마."""

from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreate(BaseModel):
    """리뷰 생성/수정 요청."""

    rating: int = Field(..., ge=1, le=5, description="전체 평점 (1-5)")
    effectiveness: int | None = Field(None, ge=1, le=5, description="효과 평점")
    ease_of_use: int | None = Field(None, ge=1, le=5, description="복용 편의성 평점")
    comment: str | None = Field(None, max_length=500, description="리뷰 코멘트")


class ReviewResponse(BaseModel):
    """리뷰 응답."""

    id: int
    device_id: str
    item_type: str
    item_id: int
    rating: int
    effectiveness: int | None = None
    ease_of_use: int | None = None
    comment: str | None = None
    helpful_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ReviewSummary(BaseModel):
    """리뷰 요약 통계."""

    average_rating: float
    total_count: int
    distribution: dict[str, int]

File: src/web/src/types/review.ts
/** 리뷰 관련 타입 정의 — 백엔드 스키마 1:1 매칭. */

export interface ReviewCreate {
  rating: number;
  effectiveness?: number | null;
  ease_of_use?: number | null;
  comment?: string | null;
  recaptcha_token?: string | null;
}

export interface ReviewResponse {
  id: number;
  device_id: string;
  item_type: string;
  item_id: number;
  rating: number;
  effectiveness: number | null;
  ease_of_use: number | null;
  comment: string | null;
  helpful_count: number;
  created_at: string;
  updated_at: string;
}

export interface ReviewSummary {
  average_rating: number;
  total_count: number;
  distribution: Record<string, number>;
}

export interface ReviewListResponse {
  items: ReviewResponse[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}
